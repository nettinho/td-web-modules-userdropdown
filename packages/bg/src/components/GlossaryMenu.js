import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { Link, withRouter } from "react-router-dom";
import { Icon, Menu } from "semantic-ui-react";
import { bindActionCreators, compose } from "redux";
import { connect } from "react-redux";
import { FormattedMessage } from "react-intl";
import { hideSidebar } from "@truedat/core/routines";
import { ActiveRoute } from "@truedat/core/components/ActiveRoute";
import { optionsDomains } from "../taxonomy/routines";
import { clearConceptQueryFilters } from "../concepts/routines";
import { clearFilterDomains } from "../taxonomy/routines";
import conceptRoutes from "../concepts/routes";
import taxonomyRoutes from "../taxonomy/routes";

export class GlossaryMenu extends React.Component {
  static propTypes = {
    hideSidebar: PropTypes.func,
    optionsDomains: PropTypes.func,
    clearConceptQueryFilters: PropTypes.func,
    clearFilterDomains: PropTypes.func,
    domainsPermissions: PropTypes.object,
    location: PropTypes.object
  };

  componentDidMount() {
    const { domainsPermissions, optionsDomains } = this.props;
    if (_.isEmpty(domainsPermissions)) {
      optionsDomains();
    }
  }

  componentDidUpdate() {
    const { clearConceptQueryFilters, clearFilterDomains } = this.props;
    const activeConcepts =
      ActiveRoute(location, conceptRoutes.CONCEPTS) ||
      ActiveRoute(location, conceptRoutes.CONCEPTS_PENDING);
    if (!activeConcepts) {
      clearConceptQueryFilters();
    }
    clearFilterDomains();
  }

  render() {
    const { hideSidebar, domainsPermissions, location } = this.props;
    const activeDomains = ActiveRoute(location, taxonomyRoutes.DOMAINS);
    const activeConcepts =
      ActiveRoute(location, conceptRoutes.CONCEPTS) ||
      ActiveRoute(location, conceptRoutes.CONCEPTS_PENDING);
    return (
      <Menu.Item
        header
        active={activeDomains || activeConcepts}
        name="glossary"
      >
        <Icon name="university" />
        <FormattedMessage id="navigation.glossary" />
        <Menu.Menu>
          {_.propEq("list", true)(domainsPermissions) && (
            <Menu.Item
              as={Link}
              to={taxonomyRoutes.DOMAINS}
              name="domains"
              active={activeDomains}
              onClick={hideSidebar}
            >
              <Icon name="sitemap" />
              <FormattedMessage id="navigation.glossary.taxonomy" />
            </Menu.Item>
          )}
          <Menu.Item
            as={Link}
            to={conceptRoutes.CONCEPTS}
            name="concepts"
            onClick={hideSidebar}
            active={activeConcepts}
          >
            <Icon name="tags" />
            <FormattedMessage id="navigation.glossary.concepts" />
          </Menu.Item>
        </Menu.Menu>
      </Menu.Item>
    );
  }
}

const mapStateToProps = ({ domainsPermissions }) => ({ domainsPermissions });

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators(
    {
      optionsDomains,
      hideSidebar,
      clearConceptQueryFilters,
      clearFilterDomains
    },
    dispatch
  )
});

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(GlossaryMenu);
