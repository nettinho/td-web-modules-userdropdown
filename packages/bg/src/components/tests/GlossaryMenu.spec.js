import React from "react";
import { shallow } from "enzyme";
import { GlossaryMenu } from "../GlossaryMenu";

describe("<GlossaryMenu />", () => {
  it("matches the latest snapshot", () => {
    const domainsPermissions = { list: true };
    const optionsDomains = jest.fn();
    const location = { pathname: "/" };

    const props = {
      domainsPermissions,
      optionsDomains,
      location
    };

    const wrapper = shallow(<GlossaryMenu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("calls optionsDomains when component mounts", () => {
    jest.spyOn(GlossaryMenu.prototype, "componentDidMount");

    const domainsPermissions = {};
    const optionsDomains = jest.fn();
    const location = { pathname: "/" };

    const props = {
      domainsPermissions,
      optionsDomains,
      location
    };

    shallow(<GlossaryMenu {...props} />);
    expect(GlossaryMenu.prototype.componentDidMount.mock.calls.length).toBe(1);
    expect(optionsDomains.mock.calls.length).toBe(1);
  });
});
