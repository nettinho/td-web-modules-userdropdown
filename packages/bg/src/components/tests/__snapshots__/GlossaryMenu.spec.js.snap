// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlossaryMenu /> matches the latest snapshot 1`] = `
<MenuItem
  active={false}
  header={true}
  name="glossary"
>
  <Icon
    as="i"
    name="university"
  />
  <FormattedMessage
    id="navigation.glossary"
    values={Object {}}
  />
  <MenuMenu>
    <MenuItem
      active={false}
      as={[Function]}
      name="domains"
      to="/domains"
    >
      <Icon
        as="i"
        name="sitemap"
      />
      <FormattedMessage
        id="navigation.glossary.taxonomy"
        values={Object {}}
      />
    </MenuItem>
    <MenuItem
      active={false}
      as={[Function]}
      name="concepts"
      to="/concepts"
    >
      <Icon
        as="i"
        name="tags"
      />
      <FormattedMessage
        id="navigation.glossary.concepts"
        values={Object {}}
      />
    </MenuItem>
  </MenuMenu>
</MenuItem>
`;
